src-exe/cabal-plan.hs

```diff
@@ -978,7 +993,32 @@ doDot showBuiltin showGlobal plan tred tredWeights highlights = either loopGraph
                 ]
         putStrLn "}"
 
-    am = planJsonDotUnitGraph plan
+    -- Start with the full graph;
+    --
+    -- If rootPatterns is not null, then find the DotUnitIds that match any
+    -- pattern given (these are the roots).
+    --
+    -- DFS from each of these roots to find the full set of DotUnitIds that we
+    -- want to keep, and restrict the graph to only these ids.
+    am :: Map DotUnitId (Set DotUnitId)
+    am =
```

Comment:
Don't filter. See how `duShow` hides builtins/globals. Similarly it can hide unreachable units, given "reachable set" is precalculated.

Reply:
also then `M.restrictKeys` (which is a function needing `containers-0.5.8`) is not required.